utils/build-script-impl

```diff
@@ -584,6 +584,12 @@ function set_build_options_for_host() {
 
             cmake_os_sysroot="$(xcrun --sdk ${platform} --show-sdk-path)"
 
+            # Give the SDK sysroot higher priority in calls to CMake `find_*`
+            # functions, over paths that may come from `pkg-config`. This fixes
+            # compile failures caused when `pkg-config` returns CommandLineTools
+            # paths instead of Xcode paths.
+            export CMAKE_PREFIX_PATH="${CMAKE_PREFIX_PATH}:${cmake_os_sysroot}/usr"
```

Comment:
Given that you append the variable, I assume the "higher priority" is achieve by setting this variable in the first place rather than it's position? I think it might be worth mentioning that or including a link to the CMake docs.

Reply:
I expanded the comment, I hope it addresses your comment, let me know if it doesn't.

With this change, there would be two ways for users to override the search paths:

1. Set a `CMAKE_PREFIX_PATH` env variable themselves, which would come before the SDK sysroot provided in this script
2. Use `-DCMAKE_PREFIX_PATH` via `--extra-cmake-options`

The link to the docs show another way of doing it per package, for example setting a `LibEdit_ROOT` environment variable.